src/level_gen.cpp

```diff
@@ -91,17 +98,22 @@ static inline Entity createAgent(Engine &ctx, const MapObject &agentInit) {
     ctx.get<EntityType>(agent) = agentInit.type;
 
     ctx.get<Goal>(agent)= Goal{.position = Vector2{.x = agentInit.goalPosition.x - ctx.data().mean.x, .y = agentInit.goalPosition.y - ctx.data().mean.y}};
-
     populateExpertTrajectory(ctx, agent, agentInit);
-
     if(!ctx.data().params.isStaticAgentControlled && (ctx.get<Goal>(agent).position - ctx.get<Trajectory>(agent).positions[0]).length() < consts::staticThreshold)
     {
         ctx.get<ResponseType>(agent) = ResponseType::Static;
     }
 
     if(ctx.data().numControlledVehicles < ctx.data().params.maxNumControlledVehicles && agentInit.type == EntityType::Vehicle && agentInit.valid[0] && ctx.get<ResponseType>(agent) == ResponseType::Dynamic)
     {
+        if (ctx.data().params.useWayMaxModel == true)
+        {
         ctx.get<ControlledState>(agent) = ControlledState{.controlledState = ControlMode::BICYCLE};
+        }
+        else
+        {
+        ctx.get<ControlledState>(agent) = ControlledState{.controlledState = ControlMode::DELTA};
```

Comment:
I think we should refactor ControlledState to a boolean. And export the BicycleModel in `mgr.cpp`.